main.py

```diff
@@ -25,7 +25,8 @@ def get_new_coins(all_coins):
     Returns new coins and the new coin list
     """
     all_coins_recheck = get_all_coins()
-    return [new_coins for new_coins in all_coins_recheck if new_coins['symbol'] not in [coin['symbol'] for coin in all_coins]], all_coins_recheck
+    new_coin_symbols = {coin['symbol'] for coin in all_coins}
+    return [new_coins for new_coins in all_coins_recheck if new_coins['symbol'] not in new_coin_symbols], all_coins_recheck
```

Comment:
This line uses that set for each new coin.

The current implementation builds the set over and over again for every new coin it sees.